### PR TITLE
dev to master (to test getting ramses)

### PR DESCRIPTION
**Pull request by ZofiaRudjord**

### CODE REVIEW

File: pyniva/request_dataframe.py [hunk 1]
```diff
@@ -109,3 +111,136 @@ def get_ship_data(
...
+    spectral_paths = [path for path in param_paths if vessel_signals[vessel_paths.index(path)].TTYPE=="spectra"]
+    assert len(spectral_paths)==1, "Only one spectral profile can be downloaded at a time"
+
+    start_datetime=datetime.strptime(start_time, "%Y-%m-%dT%H:%M:%S")
+    end_datetime=datetime.strptime(end_time, "%Y-%m-%dT%H:%M:%S")
+
+    #slide in 12 h slices
+    period =timedelta(hours=12)
+    if (end_datetime-start_datetime)>period:
+        periods = int((end_datetime-start_datetime)/period)
```
Comment (knl88) [inline pyniva/request_dataframe.py:219]: periods will be undefined if requested range is 12 hours or below so this code fails

```python
def test_get_ramses_data_6h():
    header = token2header("tests/niva-service-account.json")
    start_time = "2023-08-01T12:00:00"
    end_time = "2023-08-01T18:00:00"


    ts = get_ramses_data(
        vessel_name="FA",
        param_paths=["FA/RAMSES/RADIANCE/LD/CALIBRATED"],
        start_time=start_time,
        end_time=end_time,
        noqc=False,
        header=header,
        dt=0,
        pub_tsb="https://ferrybox.p.niva.no/v1/tsb",
        meta_host="https://ferrybox.p.niva.no/v1/metaflow"
    )


    assert not ts.empty
```

File: pyniva/request_dataframe.py [hunk 2]
```diff
@@ -109,3 +111,138 @@ def get_ship_data(
...
+    )
+    spectral_paths = [path for path in param_paths if vessel_signals[vessel_paths.index(path)].TTYPE=="spectra"]
+    assert len(spectral_paths)==1, "Only one spectral profile can be downloaded at a time"
+
+    start_datetime=datetime.strptime(start_time, "%Y-%m-%dT%H:%M:%S")
+    end_datetime=datetime.strptime(end_time, "%Y-%m-%dT%H:%M:%S")
+
+    #slide in 12 h slices
+    period =timedelta(hours=12)
+    periods = 1
```
Comment (ZofiaRudjord) [inline pyniva/request_dataframe.py:218]: Now it should be fixed, good catch!

File: pyniva/request_dataframe.py [hunk 3]
```diff
@@ -215,9 +215,11 @@ def get_ramses_data(
 
     #slide in 12 h slices
     period =timedelta(hours=12)
```
Comment (knl88) [inline pyniva/request_dataframe.py:217]: missing space:)

**Review by knl88 — APPROVED** [review]
sbu:) perhaps add a 'system' test

**Pull request merged**